### PR TITLE
Enable auto compaction for etcd

### PR DESCRIPTION
**Pull request by andmat900**
### Applicable Issues

- https://github.com/eiffel-community/etos/issues/240

### Description of the Change

This change enables automatic compaction for etcd.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Andrei Matveyeu, andrei.matveyeu@axis.com

### CODE REVIEW

File: manifests/etcd/statefulset.yaml [hunk 1]
```diff
@@ -39,7 +39,9 @@ spec:
               --initial-cluster-token etcd-cluster-1 \
               --initial-cluster ${PEERS} \
               --initial-cluster-state new \
-              --data-dir /var/run/etcd/default.etcd
+              --data-dir /var/run/etcd/default.etcd \
+              --auto-compaction-mode=revision \
+              --auto-compaction-retention=1
```
Comment (t-persson) [inline manifests/etcd/statefulset.yaml:44]: 1 hour seems extremely low when an ETOS testrun can run for about 24 hours.

File: manifests/etcd-defrag/cronjob.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+metadata:
+  name: etcd-defrag
+spec:
+  schedule: "0 0 * * *"
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          containers:
+          - name: etcd-defrag
```
Comment (t-persson) [inline manifests/etcd-defrag/cronjob.yaml:12]: Is defrag actually necessary? According to https://etcd.io/docs/v3.2/op-guide/maintenance/ :
`Compacting old revisions internally fragments etcd by leaving gaps in backend database. Fragmented space is available for use by etcd but unavailable to the host filesystem.`

File: manifests/etcd/statefulset.yaml (hunk 1 above)
Comment (andmat900) [reply manifests/etcd/statefulset.yaml:44]: The mode is set to "revision", so it is one revision, not hours.

File: manifests/etcd-defrag/cronjob.yaml (hunk 2 above)
Comment (andmat900) [reply manifests/etcd-defrag/cronjob.yaml:12]: Yes, correct. Without defragmentation the problem is the free space on the host filesystem.

> After compacting the keyspace, the backend database may exhibit internal fragmentation. Any internal fragmentation is space that is free to use by the backend but still consumes storage space. The process of defragmentation releases this storage space back to the file system.

So defragmentation is needed to free up the disk space on the host filesystem.

Comment (t-persson) [reply manifests/etcd-defrag/cronjob.yaml:12]: But the host filesystem in this case is a kubernetes volume used exclusively by etcd. Why do we need to release it?

**Review by t-persson — APPROVED** [review]

**Pull request merged**